### PR TITLE
ESP Wifi configuration

### PR DESCRIPTION
**Pull request by marekmatej**
Configure WIFI in common `kconfig.wifi`.
Introduce Wifi task priority setup.

### CODE REVIEW

**Review by LucasTambor — DISMISSED** [review]

**Review by LucasTambor — DISMISSED** [review]

File: soc/espressif/common/Kconfig.wifi [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Copyright (c) 2024 Espressif Systems (Shanghai) Co., Ltd.
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_FAMILY_ESPRESSIF_ESP32
+
+config ESP_WIFI_MAX_THREAD_PRIORITY
```
Comment (kartben) [inline soc/espressif/common/Kconfig.wifi:6]: How about adding a Wi-Fi submenu to group everything, like is done for SPIRAM?

File: soc/espressif/common/Kconfig.wifi [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	int "Maximum work queue thread priority"
```
Comment (kartben) [inline soc/espressif/common/Kconfig.wifi:7]: Make sure all descriptions have "Wi-Fi" in them (unless you feel it's redundant if you go for a menu) and spell it out consistently across descriptions/help strings.

**Review by kartben — CHANGES_REQUESTED** [review]

**Review by kartben — DISMISSED** [review]

**Review by sylvioalves — DISMISSED** [review]

File: soc/espressif/common/Kconfig.wifi [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# Copyright (c) 2024 Espressif Systems (Shanghai) Co., Ltd.
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_FAMILY_ESPRESSIF_ESP32
+
+menu "Wi-Fi config"
+
+config ESP_WIFI_MAX_THREAD_PRIORITY
```
Comment (bbilas) [inline soc/espressif/common/Kconfig.wifi:8]: I guess it's too late but it would be nicer to have `ESP32` here as well.

**Review by bbilas — DISMISSED** [review]

File: soc/espressif/common/Kconfig.wifi (hunk 3 above)
Comment (marekmatej) [reply soc/espressif/common/Kconfig.wifi:8]: you got it!

**Review by uLipe — APPROVED** [review]

**Review by kartben — APPROVED** [review]

File: soc/espressif/common/Kconfig.wifi (hunk 3 above)
Comment (bbilas) [reply soc/espressif/common/Kconfig.wifi:8]: I meant the `ESP_WIFI_MAX_THREAD_PRIORITY` option, but it requires another change in esp32 hal repo, so let's keep it.

**Review by bbilas — APPROVED** [review]

**Review by sylvioalves — APPROVED** [review]

**Pull request merged**